Python/Product/Analysis/AnalysisSet.cs

```diff
@@ -600,8 +601,8 @@ sealed class UnionComparer : IEqualityComparer<AnalysisValue>, IEqualityComparer
 #if FULL_VALIDATION
             var z2 = y.UnionMergeTypes(x, Strength);
             if (!object.ReferenceEquals(z, z2)) {
-                Validation.Assert(z.UnionEquals(z2, Strength), string.Format("{0}\n{1} + {2} => {3}\n{2} + {1} => {4}", Strength, x, y, z, z2));
-                Validation.Assert(z2.UnionEquals(z, Strength), string.Format("{0}\n{1} + {2} => {3}\n{2} + {1} => {4}", Strength, y, x, z2, z));
+                Validation.Assert(z.UnionEquals(z2, Strength), $"{Strength}\n{x} + {y} => {z}\n{y} + {x} => {z2}");
```

Comment:
If you replace `Validation.Assert` with:
```cs
        public static void Assert(bool expression, FormattableString message) {
            if (!expression) {
                try {
                    throw new ValidationException(message.ToString(CultureInfo.InvariantCulture));
                } catch (ValidationException ex) {
                    Console.Error.WriteLine(ex.ToString());
                }
            }
        }
```
you can save a lot of memory, cause `FormattableString` won't try to build a new string unless asserted expression is invalid.

Reply:
This code is disabled almost all the time - you only turn it on when looking for a particular kind of bug.